connectors/webhook/src/main/java/io/camunda/connector/inbound/authorization/JWTChecker.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. Licensed under a proprietary license.
+ * See the License.txt file for more information. You may not use this file
+ * except in compliance with the proprietary license.
+ */
+package io.camunda.connector.inbound.authorization;
+
+import com.auth0.jwk.InvalidPublicKeyException;
+import com.auth0.jwk.Jwk;
+import com.auth0.jwk.JwkException;
+import com.auth0.jwk.JwkProvider;
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.exceptions.JWTDecodeException;
+import com.auth0.jwt.exceptions.SignatureVerificationException;
+import com.auth0.jwt.exceptions.TokenExpiredException;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.camunda.connector.api.inbound.webhook.WebhookProcessingPayload;
+import io.camunda.connector.inbound.model.WebhookConnectorProperties;
+import io.camunda.connector.runtime.core.feel.FeelEngineWrapperException;
+import java.security.interfaces.ECPublicKey;
+import java.security.interfaces.RSAPublicKey;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.List;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JWTChecker {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(JWTChecker.class);
+
+  public static boolean verify(
+      final WebhookProcessingPayload payload,
+      final WebhookConnectorProperties webhookConnectorProperties,
+      final JwkProvider jwkProvider,
+      final ObjectMapper objectMapper) {
```

Comment:
Instead of passing the `webhookConnectorProperties` why not create a record object that encapsulates all these properties without the need for the dependencies. Examples:
* Instead of `webhookConnectorProperties.getRequiredPermissions()` pass the requiredPermissions as part of the record object.
* Instead of `JWTChecker.extractJWTFomHeader(payload)`  just pass the headers via the record object.

This will allow you to test this class in a more straight forward way.